sql/src/main/java/io/crate/operation/user/UserManager.java

```diff
@@ -48,4 +50,12 @@
      */
     CompletableFuture<Long> dropUser(DropUserAnalyzedStatement analysis);
 
+    /**
+     * checks if user is allowed to execute statement
+     * @param analysis          analysed statement
+     * @param sessionContext    current session context
+     * @return returns true if the session user is allowed to execute the statement
+     */
+    Boolean checkPermission(AnalyzedStatement analysis, SessionContext sessionContext);
```

Comment:
i question if it's really necessary to return something at all as you throw exceptions in the unauthorized case anyway